src/modules/pythonmonkey/pythonmonkey.cc

```diff
@@ -274,25 +274,32 @@ static bool getEvalOption(PyObject *evalOptions, const char *optionName, const c
     value = PyDict_GetItemString(evalOptions, optionName);
   }
   if (value && value != Py_None) {
-    *s_p = PyUnicode_AsUTF8(value);
+    PyObject *strObj = PyObject_Str(value); // strObj = str(value)
+    if (strObj) { // success
+      *s_p = PyUnicode_AsUTF8(strObj);
+      Py_DECREF(strObj); // PyObject_Str returns new reference
+      return true;
+    }
   }
-  return value != NULL && value != Py_None;
+  return false;
 }
 
 static bool getEvalOption(PyObject *evalOptions, const char *optionName, unsigned long *l_p) {
   PyObject *value;
   if (PyObject_TypeCheck(evalOptions, &JSObjectProxyType)) {
     value = PyMapping_GetItemString(evalOptions, optionName);
-    if (value && value != Py_None) {
-      *l_p = (unsigned long)PyFloat_AsDouble(value);
-    }
   } else {
     value = PyDict_GetItemString(evalOptions, optionName);
-    if (value && value != Py_None) {
-      *l_p = PyLong_AsUnsignedLong(value);
+  }
+  if (value && value != Py_None) {
+    PyObject *longObj = PyNumber_Long(value); // longObj = int(value)
```

Comment:
When the value is not already that type, calling `PyUnicode_AsUTF8` or `PyLong_AsUnsignedLong` will result in segfualt.

For example, I can have
```py
evalOptions = {
    'lineno': '123',
    'column': '456',
    'filename': 0,
}
```

Reply:
Interesting, I guess the code could be more robust, just like in some other parts possibly.
Obviously it has always relied on people using it as intended semantically.
Nevertheless, I told you we had a fix for the windows CI, so you should spend your time on other items, rather than finding smallish issues in here that are clearly not very important

https://blog.codinghorror.com/not-all-bugs-are-worth-fixing/
https://www.bugsnag.com/blog/not-all-bugs-are-worth-fixing/